packages/core/src/common/object.spec.ts

```diff
@@ -0,0 +1,111 @@
+            });
+            it('should deep clone strings', () => {
+                expect('foo').to.equal(deepClone('foo'));
+                expect('').to.equal(deepClone(''));
+            });
+            it('should deep clone booleans', () => {
+                expect(true).to.equal(deepClone(true));
+                expect(false).to.equal(deepClone(false));
+            });
+        });
+
+        describe('undefined', () => {
+            it('should deep clone undefined', () => {
+                expect(undefined).to.equal(deepClone(undefined));
+            });
+        });
+
+        describe('arrays', () => {
+            it('should deep clone arrays', () => {
+                const a = [1, 2, 3];
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(Array.isArray(a)).to.equal(true);
+                expect(Array.isArray(b)).to.equal(true);
+            });
+
+            it('should deep clone nested arrays', () => {
+                const a = [[1], [2]];
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(Array.isArray(a)).to.equal(true);
+                expect(Array.isArray(b)).to.equal(true);
+            });
+        });
+
+        describe('objects', () => {
+            it('should deep clone objects', () => {
+                const a = { 'foo': true, 'bar': 1 };
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(typeof a).to.equal('object');
+                expect(typeof b).to.equal('object');
+            });
+
+            it('should deep clone nested objects', () => {
+                const a = { 'foo': true, 'nested': { 'foo': 1 } };
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(typeof a).to.equal('object');
+                expect(typeof b).to.equal('object');
+            });
+        });
+    });
+
+    describe('#deepFreeze', () => {
+        it('should deep freeze an object', () => {
+            const e = { a: 10 };
+            const d = deepFreeze(e);
+            expect(Object.isFrozen(d)).to.equal(true);
+        });
+    });
+
+    describe('#notEmpty', () => {
+        it('should return "true" when not empty', () => {
+            expect(notEmpty({})).to.equal(true);
+            expect(notEmpty([])).to.equal(true);
+            expect(notEmpty({ a: 1 })).to.equal(true);
+        });
+        it('should return "false" when empty', () => {
+            expect(notEmpty(undefined)).to.equal(false);
+        });
+    });
+
+    describe('#isEmpty', () => {
+        it('should return "true" when it is Empty', () => {
+            const z = {};
+            expect(isEmpty(z)).to.equal(true);
+        });
+        it('should return "false" when not Empty', () => {
+            const z = { d: 5 };
+            expect(isEmpty(z)).to.equal(false);
+        });
+    });
```

Comment:
`isEmpty` checks two features of an object: that it has keys, and that its constructor is `Object`. These tests only confirm the behavior of one check. How can we check the `Object` constructor feature of `isEmpty`?

Reply:
Hi @colin-grant-work , 
Thanks a lot for advising about the above: I have revised my units tests according to your comment.